regression/pubsub.js

```diff
@@ -174,33 +175,98 @@ describe('pubsub', function() {
-      subscription.pull({ returnImmediately: true }, function(err, msg) {
+
+      topic.publish('hello', assert.ifError);
+
+      subscription.pull({
+        returnImmediately: true,
+        maxCount: 1
+      }, function(err, msg) {
         assert.ifError(err);
-        subscription.ack(msg.id, done);
+        subscription.ack(msg.ackId, done);
       });
-      topic.publish('hello', assert.ifError);
     });
 
     it('should receive the published message', function(done) {
       var subscription = topic.subscription(subscriptions[0].name);
-      subscription.pull({ returnImmediately: true }, function(err, msg) {
+
+      topic.publish([
+        { data: new Buffer('hello').toString('base64') },
+        { data: new Buffer('hello').toString('base64') },
+        { data: new Buffer('hello').toString('base64') },
+        { data: new Buffer('hello').toString('base64') },
+        { data: new Buffer('hello').toString('base64') },
+        { data: new Buffer('hello').toString('base64') },
+        { data: new Buffer('hello').toString('base64') },
+        { data: new Buffer('hello').toString('base64') },
+        { data: new Buffer('hello').toString('base64') },
+        { data: new Buffer('hello').toString('base64') }
+      ], assert.ifError);
+
+      subscription.pull({
+        returnImmediately: true,
+        maxCount: 1
+      }, function(err, msg) {
         assert.ifError(err);
         assert.equal(msg.data, 'hello');
-        subscription.ack(msg.id, done);
+        subscription.ack(msg.ackId, done);
       });
-      topic.publish('hello', assert.ifError);
     });
 
     it('should receive a raw published message', function(done) {
       var subscription = topic.subscription(subscriptions[0].name);
-      subscription.pull({ returnImmediately: true }, function(err, msg) {
+
+      topic.publish([
+        { data: new Buffer('hello').toString('base64') },
+        { data: new Buffer('hello').toString('base64') },
+        { data: new Buffer('hello').toString('base64') },
+        { data: new Buffer('hello').toString('base64') },
+        { data: new Buffer('hello').toString('base64') },
+        { data: new Buffer('hello').toString('base64') },
+        { data: new Buffer('hello').toString('base64') },
+        { data: new Buffer('hello').toString('base64') },
+        { data: new Buffer('hello').toString('base64') },
+        { data: new Buffer('hello').toString('base64') }
+      ], assert.ifError);
+
+      subscription.pull({
+        returnImmediately: true,
+        maxCount: 1
+      }, function(err, msg) {
         assert.ifError(err);
         assert.equal(msg.data, 'hello');
-        subscription.ack(msg.id, done);
+        subscription.ack(msg.ackId, done);
+      });
+    });
+
+    it('should receive the chosen amount of results', function(done) {
+      var subscription = topic.subscription(subscriptions[0].name);
+      var opts = { returnImmediately: true, maxCount: 3 };
+
+      topic.publish([
```

Comment:
I'm still convinced this call is race-y. Why isn't pull happening in the callback to publish?